object_manager.go

```diff
@@ -387,3 +387,20 @@ func (objMgr *ObjectManager) CreateMultiObject(req *MultiRequest) ([]map[string]
 
 	return result, nil
 }
+
+/// GetUpgradeStatus returns the status of the grid information of a map
+func (objMgr *ObjectManager) GetUpgradeStatus(type_ string) ([]map[string]interface{}, error) {
+	var res []map[string]interface{}
+
+	if type_ == "" {
+		type_ = "GROUP"
```

Comment:
According to v2.6 of WAPI doc, the only acceptable value of type is 'GRID' (please see below). Am I missing something?

```
type
    The upgrade status type.
Type
    String.
Valid values are:
    -  GRID
```

Reply:
Guess its not supported in that version, i'm trying with v2.7 & v2.8.

here's the  latest:

> 
> type
> The type of upper level elements to be requested.
> 
> Type
> 
> String.
> 
> Valid values are:
> GRID
> GROUP
> PNODE
> VNODE
> 

So should I default it to "GRID" ?

Reply:
Thanks for the info! In this case, I suggest the following:

1) Considering the nature of the call, I wouldn't even provide a default. It should just validate the provided value against an acceptable set. (i.e. An empty string would be an error.)

2) More importantly, since the set of acceptable values dependent on the WAPI version, we need to take that into account. The python version has a `Feature` class (here: [https://github.com/infobloxopen/infoblox-client/blob/master/infoblox_client/feature.py](url)) that is designed to support this sort of situation. Please take a look and consider implementing something like that here.